### PR TITLE
New folder for 5.3.1 / fixed dev app

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: common/ViewController60.swift [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                self.updateUI();
+                Thread.sleep(forTimeInterval:0.2);
+            }
+            self.running = false;
+            self.updateUI();
+        };
+    }
+    
+    func getCurrentCounterFromDb() -> Int {
+        var result = self.store.query(with:SFQuerySpec.newSmartQuerySpec("SELECT count(*) FROM {\(SOUP_NAME)}", withPageSize: 1)!, pageIndex:0, error:nil);
```
Comment (wmathurin) [inline common/ViewController60.swift:96]: A few changes were required to have it compile in 6.0

**Comment by wmathurin** [comment]
Note: Working on upgrade testing from 5.3.1 to 6.0 and it's not working :-( It's not a sqlcipher issue, the key used to attempt unlocking is different in 6.0.

**Comment by bhariharan** [comment]
We decoupled the encryption key from passcode. I thought we tested upgrade?

**Comment by wmathurin** [comment]
The problem is here: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/master/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFKeyStoreManager.m#L146

Keys of type passcode are stored in the generated store if there is no passcode in use. 

My upgrade code takes care of moving the keys from the passcode store to the generated store as soon as the passcode is known.
I need to take care of keys labelled with the passcode suffix in the generated store.

**Review by bhariharan — APPROVED** [review]

**Comment by wmathurin** [comment]
Fixing the issue here: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2314

**Pull request merged**